embassy-nrf/src/wdt.rs

```diff
@@ -178,25 +189,50 @@ impl WatchdogHandle {
     /// prevent a reset from occurring.
     #[inline]
     pub fn pet(&mut self) {
-        let r = crate::pac::WDT;
-        r.rr(self.index as usize).write(|w| w.set_rr(vals::Rr::RELOAD));
+        let r = self.regs();
+        r.rr(self.rr_index()).write(|w| w.set_rr(vals::Rr::RELOAD));
     }
 
     /// Has this handle been pet within the current window?
     pub fn is_pet(&self) -> bool {
-        let r = crate::pac::WDT;
-        !r.reqstatus().read().rr(self.index as usize)
+        let r = self.regs();
+        !r.reqstatus().read().rr(self.rr_index())
     }
 
     /// Steal a watchdog handle by index.
     ///
     /// # Safety
-    /// Watchdog must be initialized and `index` must be between `0` and `N-1`
-    /// where `N` is the handle count when initializing.
-    pub unsafe fn steal(index: u8) -> Self {
+    /// Watchdog must be initialized, `instance_index` must be the index
+    /// [`Instance::INDEX`], and `index` must be between `0` and `N-1` where `N`
+    /// is the handle count when initializing.
+    pub unsafe fn steal(instance_index: u8, index: u8) -> Self {
         Self {
-            _wdt: PhantomData,
-            index,
+            index: instance_index * 8 + index,
         }
     }
 }
+
+pub(crate) trait SealedInstance {
+    const REGS: pac::wdt::Wdt;
+}
+
+/// WDT instance.
+#[allow(private_bounds)]
+pub trait Instance: SealedInstance + PeripheralType + 'static + Send {
+    /// Interrupt for this peripheral.
+    type Interrupt: interrupt::typelevel::Interrupt;
+    /// Index of the watchdog instance.
+    const INDEX: u8;
```

Comment:
can you put `INDEX` in `SealedInstance`, so we don't make it public?

Reply:
I had it public because `WatchdogHandle::steal` is also public and it takes the instance index. Actually I could make `steal` take the instance as a type parameter instead.

Reply:
ah right . yeah that works.